internal/k8s/statefulset/statefulset.go

```diff
@@ -92,7 +97,7 @@ func (r *reconciler) Reconcile(req reconcile.Request) (res reconcile.Result, err
 		return
 	}
 
-	ssm := make(map[string][]StatefulSet)
+	ssm := r.pool.Get().(map[string][]StatefulSet)
 	appList := make(map[string]bool)
 
 	for _, statefulset := range ssl.Items {
```

Comment:
For L104: I wonder why the `name` is derived from "app" labels. I think it is simpler that it is derived from the name of StatefulSet.